Framework/API/src/Algorithm.cpp

```diff
@@ -584,6 +592,13 @@ bool Algorithm::execute() {
       setExecuted(true);
 
       // Log that execution has completed.
+      getLogger().debug() << "Time to validate properties: "
```

Comment:
We've observed in the past that there is quite a difference in performance with the loggers if you build the string up externally, either with a `stringstream` or just a `string` and `to_string` and then pass it to getLogger().debug(message_str) in one shot. Is that worth doing here?